mod/mindbreeze_test/start.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+elgg_register_event_handler('init', 'system', 'mindbreeze_test_init');
+
+function mindbreeze_test_init() {
+
+    elgg_register_plugin_hook_handler('route', 'all', 'mindbreeze_route_all', 10);
+
+}
+
+/**
+ * Handles incoming HTTP Request from the GSA, this acts as the security layer for the GSA only
+ * @param $hook
+ * @param $type
+ * @param $info
+ * @return False to stop processing the response, True will let elgg handle the response
+ */
+function mindbreeze_route_all($hook, $type, $info) {
+
+    // Incoming request from the GSA
+    $agent = strtolower($_SERVER['HTTP_USER_AGENT']);
+    $set_agent = strtolower(elgg_get_plugin_setting('elgg_mb_agentstring', 'mindbreeze_test'));
+
+    if (strpos($agent, $set_agent) !== false) {
+  
+  		$set_username = strtolower(elgg_get_plugin_setting('elgg_mb_username','mindbreeze_test'));
```

Comment:
the plugin mindbreeze won't be part of GCconnex (long story short, this was for a different project that i did not have access to...)